Hybrid studies : bounding unsup energy variable from modeler [ANT-4033] (#3382)
AntaresSimulatorTeam/Antares_Simulator: docs/Architecture_Decision_Records/from-GEMS-to-legacy-linear-preblem.md

# Adding expressions from GEMS into legacy LP constraints

> Every time we want to add an expressions from GEMS into an existing legacy LP constaint, this expression 

Reviewer: ```suggestion
Every time we want to add an expression from GEMS into an existing legacy LP constaint, this expression 
```

Author: Done

is evaluated into a time-dependent linear expression.
A time-dependent linear expression is a vector of linear expressions (one for each time step).

> A linear expression is comprises a linear combination of variables and a unique constant term.

Reviewer: ```suggestion
A linear expression is comprised of a linear combination of variables and a unique constant term.
```

Author: I could have made a mistake, but not this time : 
example from worldreference.com : 
**The metropolitan area comprises downtown and several suburbs.**

Reviewer 2: Yes so you need to remove the `is`

Author: oops, sorry

Author: Done

In a linear expression, each variable is associated to a constant coefficient.

So when adding an expression from GEMS into a LP constraint, we merely add coefficients or constants to the constraint.

> In the following, writting constraints often implies gathering constant-in-time terms in the RHS, 

Reviewer 2: ```suggestion
In the following, writing constraints often implies gathering constant terms in the RHS, 
```
The RHS gathers constants (as opposed to variable in the optimization sense) but these constant can be different at each time step

Author: ok, done.

and variable terms to the LHS.

## Balance constraint

In Antares legacy modeler code, balance constraint is actually written as follows :
 **-P = -(L-Fatal)**, with **P** being productions, **L** being loads and **L-Fatal** being the residual load.
An equivalent form is : **L - P = Fatal**.

> Now, let's break P into Pv (production variable in time) and Pc (production constant in time) :

Reviewer 2: ```suggestion
Now, let's break P into Pv (production given as a variable to the optimization) and Pc (production that is a parameter, e.g. fatal production) :
```

Author: Done

**L - Pv = Fatal + Pc**
As a consequence, variable productions from GEMS are added negatively, and constant productions are added positively. 

## Fictitious load constraint

This constraint can be written as follows : 

**Spillage - Thermal - Hydro < Fatal**.  

> As previously, adding time-variable productions from GEMS is done negatively, and adding time-constant production is done positively.

Reviewer 2: ```suggestion
As previously, adding variable productions from GEMS is done negatively, and adding constant production is done positively.
```

Author: Done

## Bounding unsupplied enegy constraint

This constraint can be written as follows :

**unsupE - Lv < Lc + Fatal**, **Lv** being variables loads, **Lc** being constant loads.

> adding time-variable loads from GEMS is done negatively, and adding time-constant loads is done positively.

Reviewer 2: I would add in another paragraph : in the GEMS models, the convention is to count productions positively for:
-  the balance constraint contribution (from `injection-to-balance`)
- the fictitious load constraint contribution (from `spillage-bound`)

And to count loads positively for the unsupplied energy constraint contribution (from `unsupplied-energy-bounds)

In all three cases, when reading the legacy way of writing the constraint, we should therefore take `-coef` (link to line of code ?) of the GEMS LinearExpression to complete the legacy constraint and `+constant`

Author: Done in users file **08-hybrid-studies.md**
